osu.Game/Screens/Edit/EditorClock.cs

```diff
@@ -25,8 +25,13 @@ public class EditorClock : Component, IFrameBasedClock, IAdjustableClock, ISourc
 
         private readonly DecoupleableInterpolatingFramedClock underlyingClock;
 
-        public EditorClock(WorkingBeatmap beatmap, BindableBeatDivisor beatDivisor)
-            : this(beatmap.Beatmap.ControlPointInfo, beatmap.Track.Length, beatDivisor)
+        public EditorClock(WorkingBeatmap beatmap, double trackLength, BindableBeatDivisor beatDivisor)
```

Comment:
This feels a little awkward in that `trackLength` is passed *alongside* the beatmap. Can't the track length be accessed by just doing `beatmap.GetTrack().Length`?